ios/FluentUI/Vnext/Notification/NotificationView_SwiftUI.swift

```diff
@@ -45,41 +42,98 @@ import SwiftUI
 /// View that represents the Notification.
 public struct NotificationViewSwiftUI: View, ConfigurableTokenizedControl {
     @Environment(\.horizontalSizeClass) var horizontalSizeClass: UserInterfaceSizeClass?
-    @Environment(\.swiftUIInsets) private var safeAreaInsets: EdgeInsets
     @Environment(\.fluentTheme) var fluentTheme: FluentTheme
     @ObservedObject var state: MSFNotificationStateImpl
+    @Binding var isPresented: Bool
+    @State private var dismissedOffset: CGFloat = 0
+    @State private var yOffsetFromBottom: CGFloat = 0
     let defaultTokens: NotificationTokens = .init()
     var tokens: NotificationTokens {
         let tokens = resolvedTokens
         tokens.style = state.style
         return tokens
     }
+
     public func overrideTokens(_ tokens: NotificationTokens?) -> NotificationViewSwiftUI {
         state.overrideTokens = tokens
         return self
     }
 
     public init(style: MSFNotificationStyle,
                 message: String,
-                delayTime: TimeInterval) {
-        let state = MSFNotificationStateImpl(style: style, message: message, delayTime: delayTime)
+                isPresented: Binding<Bool>? = nil,
+                title: String = "",
+                image: UIImage? = nil,
+                actionButtonTitle: String = "",
+                actionButtonAction: (() -> Void)? = nil,
+                messageButtonAction: (() -> Void)? = nil,
+                dismissAction: (() -> Void)? = nil) {
+        let state = MSFNotificationStateImpl(style: style, message: message)
+        state.title = title
+        state.image = image
+        state.actionButtonTitle = actionButtonTitle
+        state.actionButtonAction = actionButtonAction
+        state.messageButtonAction = messageButtonAction
+        state.dismissAction = dismissAction
         self.state = state
-    }
-
-    private var hasImage: Bool {
-        state.style.isToast && state.image != nil
-    }
 
-    private var hasSecondTextRow: Bool {
-        state.style.isToast && state.title != ""
+        if let isPresented = isPresented {
+            _isPresented = isPresented
+        } else {
+            _isPresented = .constant(true)
+        }
     }
 
-    private var hasCenteredText: Bool {
-        !state.style.isToast && state.actionButtonAction == nil
+    public var body: some View {
+        GeometryReader { geometryReader in
+            let width = geometryReader.size.width
+            innerContents
+                .onTapGesture {
+                    if let messageAction = state.messageButtonAction, let dismissAction = state.dismissAction {
+                        isPresented = false
+                        dismissAction()
+                        messageAction()
+                    }
+                }
+                .frame(width: state.style.isToast && horizontalSizeClass == .regular ? width / 2 : width - (2 * tokens.presentationOffset), alignment: .bottom)
```

Comment:
Is there a way to make the width param more concise and easier to read?

I'm one way I can think of is to initialize the width constant in line 89 with a closure and have all this logic there.

```
            let width: CGFloat = {
                ...
            }()
```

Thoughts?

Reply:
Looks good to me!